### DISCUSSION TITLE
Please help me to modify the frequency font

### DISCUSSION THREAD

**Discussion by leryfm** [PE5PVB/TEF6686_ESP32#409] [Q&A]
Hello dear Sjef, and other respectal members of the TEF's firmware community! 

Thank you for creating such wonderful firmware for the Tef receiver! This is a very interesting device that inspired and interested me.
Let me start an important topic of discussion.

Recently, the **font**, which displaying the frequency of the radio station has been changed in the firmware to new one.

**I read here, that returning old font to the current firmware is impossible. It's a pity. :(**

I would like to use it at least for myself. I know how to flash a radio not through a bin file, but through the Arduino IDE program, using the appropriate instruction. This method allows to load firmware with any of my own changes into the receiver.

But i don't know, where the previous "segment" font was located in the source code?

As I discovered, a new font now located in **FREQFONT.h** file. But the previous version of source code (dated ~august 2023) for the firmware that contained the old font instead of the new one **does not have such file**. 
Accordingly, I cannot take an old file and insert it into a new source code, obtaining the current functionality, but with the old frequency segment font. Apparently it won't work that way. 

Please, can you figure out, how to get it out of the old firmware if the FREQFONT.h file is missing there. I couldn't find it in other files, how can I do this correctly? 

**WHY IS IT NEEDED**: I think that this is a tribute to the HISTORY of radio as such. Many radio receivers, of different years of production, which contained a frequency display (stationary tuners, cassette- or cd-players with radio, old "non-android" auto-radios, etc), used the exactly SEGMENT font for frequency, not any other. This created a certain aesthetic for this font. 
Many radio frequencies are associated with this for me, and I think for many other users too. It is aestetically important to see the frequency in the style of segment numbers when I receive different radio stations, not the any another font. 
**I can call it "the fundamental visual part of the radio experience".**

A new font is certainly also beautiful, but this moment is a little frustrating now, I want to simultaneously use the new functionality of new firmware versions, but see the frequency exactly in segments, in the old way, as it was until August 2023.

_Maybe there is a chance to still ask you to return the old font to firmware code, along WITH the new one (and make a choice in the menu, like choosing a setting point)? (I have already read a negative response to this request, but maybe there is still some chance to convince you to change your mind? :) I really hope the device's memory can store both fonts and that they do not take up much place)_

This problem is important for me. Because of this, i have to be torn between the old firmware, when i just want to enjoy of the radio, and the new one, when i want to try out the new functionality, but “tolerate” the new font. :( 

I hope for any your assistance in this. Ideally, return the font to the firmware (in no case instead of a new font, it is better to save both for the user’s choice), or at least explain how to replace it by myself in the source code, if it is fundamentally impossible to return... 

Thank you! Best Regards! 
Evgenij

**Comment by andimik** [comment]
BTW: See closed https://github.com/PE5PVB/TEF6686_ESP32/issues/305

> As I discovered, a new font now located in FREQFONT.h file. But the previous version of source code (dated ~august 2023) for the firmware that contained the old font instead of the new one does not have such file.

No, that's not true. You always can use any commit from the past. If you want to change the source code:

https://github.com/PE5PVB/TEF6686_ESP32/commit/ab94e952dd51d416750b12a3aac5b08447f7cd6d

This is the old file: https://github.com/PE5PVB/TEF6686_ESP32/blob/ab94e952dd51d416750b12a3aac5b08447f7cd6d/src/FREQFONT.h

**Comment by andimik** [comment]
But don't waste your time. This will not work, I just have tried it.

If you want to have the old font back, you simply can install any beta prior to this date.

**Comment by PE5PVB** [comment]
The old segment font comes with the tft_espi library. It's too complicated to make is selectable, but since this is an open source project you can fork the firmware and change it for yourself.

**Comment by leryfm** [comment]
Thank you very much for your answers. OK, I'll try to do it myself. Tell me, what was the name of this "segmental" font in the tft_espi library? I have read about it, there are a lot of fonts, including non-segment ones. How to find this correctly in it by name?

**Comment by PE5PVB** [comment]
The fonts are numbered. I thought it was number 4. But you can look it up at the older v1.15 software.

**Comment by leryfm** [comment]
Thank you! Thank you! Thank you! 

In fresh version of firmwre i found the new fonts!!! This is really BEST New Year's Gift for me! :))) 

A "Classic" named font (segmental) is a really what i needed! I would also like to note "Comic" font, it is really funny and brings a good mood!

"Roubenstil" - very stylish and cool too! I once saw this font (or a very similar) in the form of a frequency on a radio station I liked a long time ago. 

"Motoya" - also beautiful and seemed interesting. I will now use all these fonts in turn, it is nice and important.

Pleasure. Thanks!!! Merry Christmas and a Happy New Year!!!)

**Comment by PE5PVB** [comment]
You're welcome.
